addons/mail/static/src/chatter/web/form_compiler.js

```diff
@@ -93,24 +93,14 @@ patch(FormCompiler.prototype, {
             });
         }
         // after sheet bg (standard position, either aside or below)
-        if (webClientViewAttachmentViewHookXml) {
-            setAttributes(chatterContainerHookXml, {
-                "t-if": `!(__comp__.hasFileViewer() and __comp__.uiService.size >= ${SIZES.XXL})`,
-                "t-attf-class": `{{ __comp__.uiService.size >= ${SIZES.XXL} and !(__comp__.hasFileViewer() and __comp__.uiService.size >= ${SIZES.XXL}) ? "o-aside" : "" }}`,
-            });
-            setAttributes(chatterContainerXml, {
-                isInFormSheetBg: "__comp__.hasFileViewer()",
-                isChatterAside: `__comp__.uiService.size >= ${SIZES.XXL} and !(__comp__.hasFileViewer() and __comp__.uiService.size >= ${SIZES.XXL})`,
-            });
-        } else {
-            setAttributes(chatterContainerXml, {
-                isInFormSheetBg: "false",
-                isChatterAside: `__comp__.uiService.size >= ${SIZES.XXL}`,
-            });
-            setAttributes(chatterContainerHookXml, {
-                "t-attf-class": `{{ __comp__.uiService.size >= ${SIZES.XXL} ? "o-aside" : "mt-4 mt-md-0" }}`,
-            });
-        }
+        setAttributes(chatterContainerXml, {
+            isInFormSheetBg: `["combo", "bottom_chatter"].includes(__comp__.mailLayout(${hasPreview}))`,
+            isChatterAside: `["side_chatter", "external_combo_xxl", "external_combo"].includes(__comp__.mailLayout(${hasPreview}))`,
+        });
+        setAttributes(chatterContainerHookXml, {
+            "t-if": `["side_chatter", "external_combo", "external_combo_xxl", "bottom_chatter"].includes(__comp__.mailLayout(${hasPreview}))`,  // opposite of sheetBgChatterContainerHookXml
```

Comment:
If `sheetBg` is `layout == "combo"` and that's the opposite, couldn't this just be `layout != "combo"`?

Reply:
Well, there is also `none` if there is no chatter and no attachment

Reply:
Then could be `t-if="!none and !combo"` so we don't have to diff this LOC each time we adapt the states.